http-client/src/main/java/io/airlift/http/client/spnego/SpnegoAuthentication.java

```diff
@@ -103,6 +107,21 @@ public URI getURI()
 
             @Override
             public void apply(Request request)
+            {
+                try {
```

Comment:
Using `failsafe` would look something like this:
```java
RetryPolicy<?> retryPolicy = new RetryPolicy<>()
        .withBackoff(1, 5, ChronoUnit.SECONDS)
        .withMaxRetries(5)
        .handle(RuntimeException.class)
        .abortIf((result, e) -> !networkIssue(e));
        .onRetry(event -> log.debug(
                "Authentication failed on attempt %s, will retry. Exception: %s",
                event.getAttemptCount(),
                event.getLastFailure().getMessage()));

try {
    FailSafe.with(retryPolicy).run(() -> authenticate(request));
}
catch (FailsafeException e) {
    throwIfUnchecked(e.getCause());
    throw e; // should not happen
}

Reply:
Catch of `FailsafeException` is not needed because method `run`:

>  `@throws` FailsafeException if the {`@code` supplier} fails with a checked Exception or if interrupted while waiting to perform a retry.FailsafeException if the {`@code` supplier} fails with a checked Exception or if interrupted while waiting to perform a retry.